buffer/src/main/java/io/netty/buffer/SlicedAbstractByteBuf.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer;
+
+/**
+ * A special {@link SlicedByteBuf} that can make optimizations because it knows the sliced buffer is of type
+ * {@link AbstractByteBuf}.
+ */
+final class SlicedAbstractByteBuf extends SlicedByteBuf {
```

Comment:
nit: The name is a little odd since it's final but contains `Abstract` in the name  :) ... maybe just make this a private inner class of `AbstractByteBuf`?